File: Examples/test-suite/javascript/smart_pointer_multi_runme.js
var smart_pointer_multi = require("smart_pointer_multi");

var f = new smart_pointer_multi.Foo();
var b = new smart_pointer_multi.Bar(f);
var s = new smart_pointer_multi.Spam(b);
var g = new smart_pointer_multi.Grok(b);

s.x = 3;
if (s.getx() != 3) {
    throw new Error;
}

g.x = 4;
if (g.getx() != 4) {
    throw new Error;
}

File: Examples/test-suite/lua/smart_pointer_multi_runme.lua
require("import")
require("smart_pointer_multi")
spm=smart_pointer_multi
catch_undef_globs() -- catch "undefined" global variables

foo = spm.Foo()
foo.x = 5
assert(foo:getx() == 5)

bar = spm.Bar(foo)
spam = spm.Spam(bar)
grok = spm.Grok(bar)

assert(bar:getx() == 5)
assert(spam:getx() == 5)
spam.x = 7
assert(grok:getx() == 7)
grok.x = 10
assert(foo:getx() == 10)
